engine/client/client.go

```diff
@@ -322,9 +322,11 @@ func (c *Client) startEngine(ctx context.Context) (rerr error) {
 	if c.EngineCallback != nil {
 		c.EngineCallback(ctx, bkInfo.BuildkitVersion.Revision, bkInfo.BuildkitVersion.Version, c.ID)
 	}
-	if c.CloudCallback != nil {
+	if c.CloudURLCallback != nil {
 		if url, msg, ok := enginetel.URLForTrace(ctx); ok {
-			c.CloudCallback(ctx, url, msg)
+			c.CloudURLCallback(ctx, url, msg, ok)
+		} else {
+			c.CloudURLCallback(ctx, "https://dagger.cloud/traces/setup", "", ok)
```

Comment:
nit: I'm 50/50 on this, would have expected that we don't call the callback and just handle it being unset when you press the hotkey. Seems like a possible footgun later to have a URL that can mean two different things, e.g. phrasing around it might be like "View your trace in Cloud: <signup url>"

But on the other hand maybe it's nice to distinguish between "I don't know if I'm logged in" vs. "I know I am, here's the trace URL" vs. "I know I'm not, here's the signup URL."

:man_shrugging:

Reply:
> But on the other hand maybe it's nice to distinguish between "I don't know if I'm logged in" vs. "I know I am, here's the trace URL" vs. "I know I'm not, here's the signup URL.

That's exactly the same feeling I had while doing it. We'll always have a URL now regardless is the user is logged in or not, that's why I also renamed the function to `SetCloudURL` instead of the former `ConnectedToCloud` name which was only used in "logged in" cases.